src/main/java/com/alvarium/annotators/vulnerability/GoModHandler.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators.vulnerability;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class GoModHandler implements PackageFileHandler {
+    final private File file;
+
+    protected GoModHandler(String dir) {
+        this.file = new File(dir + "/" + "go.mod");
+    }
+
+    @Override
+    public String getFileName() {
+        return this.file.getName();
+    }
+
+    @Override
+    public Map<String, String> getPackages() throws VulnerabilityException {
+        Map<String, String> packages = new HashMap<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
+            boolean inRequireBlock = false;
+            boolean inReplaceBlock = false;
+            while ((line = reader.readLine()) != null) {
+                line = line.trim();
+                if (this.isRequireBlock(line)) {
+                    inRequireBlock = true;
+                    continue;
+                }
+                if (this.isReplaceBlock(line)) {
+                    inReplaceBlock = true;
+                    continue;
+                }
+                if (line.startsWith(")")) {
+                    inRequireBlock = false;
+                    inReplaceBlock = false;
+                    continue;
+                }
+                if (inRequireBlock) {
+                    String[] parts = line.split("\\s+");
+                    if (parts.length >= 2) {
```

Comment:
Just out of curiosity, is this condition to ensure that this isn't an empty line ?

Reply:
Yes exactly